packages/ui-components/src/components/NatAmountInput.jsx

```diff
@@ -33,17 +37,47 @@ const makeNatAmountInput = ({ React, TextField }) => ({
     }
   };
 
+  // Returns a function, that, as long as it continues to be invoked, will not
+  // be triggered. The function will be called after it stops being called for
+  // `wait` milliseconds.
+  const debounce = (func, wait) => {
+    let timeout;
+
+    return function executedFunction(...args) {
+      const later = () => {
+        clearTimeout(timeout);
+        func(...args);
+      };
+
+      clearTimeout(timeout);
+      timeout = setTimeout(later, wait);
+    };
+  };
+
+  const delayedOnChange = debounce(str => {
+    onChange(parseAsNat(str, decimalPlaces));
+  }, 50);
```

Comment:
put the magic number in a constant: DEBOUNCE_WAIT_MS

Reply:
Yup!